torch/csrc/jit/codegen/cuda/scheduler.cpp

```diff
@@ -313,22 +313,23 @@ ReductionParams reductionHeuristic(
       at::cuda::getCurrentDeviceProperties()->multiProcessorCount;
 
   int blocks_per_sm = device_max_threads_per_multiprocessor /
-      (rparams.block_dim_x_ * rparams.block_dim_y_);
+      (rparams.bdimx_.value_ * rparams.bdimy_.value_);
   int target_grid_size = device_multiprocessor_count * blocks_per_sm;
 
   // Setting the number of blocks based on the number of outputs
-  rparams.grid_dim_x_ = ceilDiv(red_outputs, outputs_produced_per_block_iter);
+  rparams.gdimx_.value_ = ceilDiv(red_outputs, outputs_produced_per_block_iter);
+  rparams.gdimx_.mutable_ = true;
 
   // Cross-block reductions (if necessary)
   if (rparams.cross_block_ && red_elems_per_thread >= kMaxValuesPerThread &&
-      rparams.grid_dim_x_ <= target_grid_size) {
-    int blks_per_out_1 = ceilDiv(target_grid_size, rparams.grid_dim_x_);
+      rparams.gdimx_.value_ <= target_grid_size) {
+    int blks_per_out_1 = ceilDiv(target_grid_size, rparams.gdimx_.value_);
     int blks_per_out_2 = ceilDiv(red_elems_per_thread, kMinValuesPerThread);
     int blks_per_out_3 = ceilDiv(red_elems_per_thread, kMaxValuesPerThread);
     int blks_per_output =
         std::max(std::min(blks_per_out_1, blks_per_out_2), blks_per_out_3);
 
-    rparams.grid_dim_y_ = std::max(1, blks_per_output);
+    rparams.gdimy_.value_ = std::max(1, blks_per_output);
```

Comment:
dumb question, `gdimx_` is marked mutable. Here `gdimy_` could be dependent on `gdimx_`, but is not marked `mutable`?

Maybe I just missed some logic.